Src/Test.UnitTests.Sarif.PatternMatcher/ReflectionTests.cs

```diff
@@ -55,5 +55,83 @@ public static void VerifyAllValidatorsExist(string definitionsFileName, Assembly
                 Assert.True(false, "Unexpected JSON structure in rules file");
             }
         }
+
+        public static void VerifyAllSharedStringsExist(string definitionsFilePath, string sharedStringsFilePath)
+        {
+            try
+            {
+                if (!File.Exists(definitionsFilePath) || !File.Exists(sharedStringsFilePath))
+                {
+                    return;
+                    // Should we Assert.True(false) here?
+                }
+
+                string content = File.ReadAllText(definitionsFilePath);
+
+                var jObject = (JObject)JToken.Parse(content);
+
+                IEnumerable<string> rules = jObject["Definitions"][0]["MatchExpressions"].Select(x => x["ContentsRegex"].ToString()).Distinct();
+
+                var rulesWithoutSharedStrings = new List<string>();
+                string sharedStringsContents;
+                using (var sr = new StreamReader(sharedStringsFilePath))
+                {
+                    sharedStringsContents = sr.ReadToEnd();
+                }
+
+                foreach (string rule in rules)
+                {
+                    if (!sharedStringsContents.Contains(rule))
+                    {
+                        rulesWithoutSharedStrings.Add(rule);
+                    }
+                }
+
+                // Assert.Empty doesn't allow custom messages, so use Assert.True
+                Assert.True(rulesWithoutSharedStrings.Count == 0, "Unable to find shared strings for these rules: " + string.Join(',', rulesWithoutSharedStrings));
```

Comment:
>rulesWithoutSharedStrings [](http://example.com/codeflow?start=28&length=25)

this line of code is too long, you should start to develop a style for breaking this sort of thing across multiple lines.

doing that somehow encourages creating utility method (which I find a helpful prompt).